docs/node/mainnet/e-upgrade.md

```diff
@@ -100,3 +100,310 @@ upgrade. For now, please take a look at the related [Testnet upgrade].
+compatible genesis file.
+
+The new genesis file will be published on the above link soon after reaching the
+upgrade epoch.
+
+:::
+
+3. Verify the provided Mainnet upgrade genesis file by comparing it to the
+   local network state dump.
+
+   Find the `genesis-mainnet-XXX.json` file in the `exports` subdirectory in
+   your data dir (e.g. `/node/`, `/srv/oasis/node/`)  and run `sha256sum` on it.
+   Afterwards, compare it with the hash that we will share on the
+   `#node-operators` Discord channel.
+
+   The state changes are described in the [State Changes](#state-changes)
+   section below.
+
+4. Replace the old genesis file with the new Mainnet genesis file.
+
+5. Ensure your node will remain stopped by disabling auto-starting via your
+   process manager (e.g., [systemd] or [Supervisor])
+
+6. Back up the entire data directory of your node. Verify that the backup
+includes the following folders:
+- for consensus: `tendermint/abci-state` and `tendermint/data`
+- for runtimes: `runtimes/*/mkvs_storage.badger.db` and
+`runtimes/*/worker-local-storage.badger.db`
+
+7. [Wipe state]. This must be performed _before_ replacing the Oasis Node
+   binary. In case you are upgrading ParaTimes/runtimes ensure you read the
+   following section:
+
+:::danger
+
+State of ParaTimes/runtimes is not affected by this upgrade and MUST NOT be
+wiped. Wiping state for confidential ParaTimes will prevent your compute or
+key manager node from transitioning to the new network.
+
+To safely wipe the blockchain state on a runtime while preserving the runtime
+state, follow these steps:
+1. **Dry Run:** Initiate a dry run to preview which files will be deleted
+by running the following command: 
+```
+oasis-node unsafe-reset --dry_run
+```
+2. **Wipe blockchain state:** After reviewing the dry run results, proceed with
+the reset by running:
+```
+oasis-node unsafe-reset
+```
+
+Transitioning confidential ParaTimes to the new network requires local state
+that is sealed to the CPU. This also means that bootstrapping a new node on a
+separate CPU immediately after the network upgrade will not be possible until
+an updated ParaTime containing new trust roots is released and adopted.
+
+:::
+
+8. Replace the old version of Oasis Node with version [23.0.x].
+
+:::info
+
+The Oasis Core 23.0.x binary in our published releases is built only for Ubuntu
+22.04 (GLIBC>=2.32). You'll have to build it yourself if you're using prior
+Ubuntu versions (or other distributions using older system libraries).
+
+:::
+
+9. Perform any needed [configuration changes](#configuration-changes) described
+   below.
+
+10. (only Rosetta Gateway operators) Replace old version of Oasis Rosetta
+    Gateway with version [2.6.0][rosetta-gw-2.6.0].
+
+11. Start your node and re-enable auto-starting via your process manager.
+
+[Oasis CLI]: ../../general/manage-tokens/cli/
+[Import your keys into the wallet]: ../../general/manage-tokens/cli/wallet/#import
```

Comment:
This should be the right path once https://github.com/oasisprotocol/cli/pull/146 lands.

```suggestion
[Import your keys into the wallet]: ../../general/manage-tokens/cli/wallet.md#import-file
```

Reply:
Thanks! Link updated.